packages/web-components/@sfx/search-box/src/search-box.ts

```diff
@@ -0,0 +1,205 @@
+import { Base } from '@sfx/base';
+import { SEARCHBOX_EVENT, KEY_CODES } from './events';
+
+/**
+ * This entity is responsible for receiving user input and dispatching events
+ * based on input.
+ * The entity also listens for events, and updates data accordingly.
+ */
+@customElement('sfx-search-box')
+export default class SearchBox extends Base {
+  /**
+   * Text used as placeholder in search box.
+   */
+  @property({ type: String })
+  placeholder: string = 'Type your search';
+  /**
+   * Search term generated via user input.
+   */
+  @property({ type: String }) value: string = '';
+  /**
+   * Determines whether or not the search button is present.
+   */
+  @property({ type: Boolean, reflect: true })
+  searchButton: boolean = false;
+  /**
+   * Determines whether or not the clear button is present.
+   */
+  @property({ type: Boolean, reflect: true })
+  clearButton: boolean = false;
+
+  constructor() {
+    super();
+    this.updateText = this.updateText.bind(this);
+  }
+
+  /**
+   * Adds event listeners.
+   */
+  connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener(SEARCHBOX_EVENT.UPDATE_SEARCH_TERM, this.updateText);
+  }
+
+  /**
+   * Removes event listeners.
+   */
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener(SEARCHBOX_EVENT.UPDATE_SEARCH_TERM, this.updateText);
+  }
+
+  /**
+   * Dispatches a search request event with the `value` property.
+   * Invoked in response to user interactions: `enter` key or click on search button.
+   */
+  emitSearchEvent() {
+    const searchboxRequestEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCH_REQUEST, {
+      detail: this.value,
+      bubbles: true
+    });
+    this.dispatchEvent(searchboxRequestEvent);
+  }
+
+  /**
+   * Dispatches an event notifying that the input box has been cleared.
+   * Invoked in response to a click on the clear button.
+   */
+  emitSearchBoxClearClick() {
+    const searchboxClearedEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CLEAR_CLICK, { bubbles: true });
+    this.dispatchEvent(searchboxClearedEvent);
+  }
+
+  /**
+   * Updates the contents of the search input box and the value property
+   * with the payload of the given event.
+   * Invoked in response to an update search term event.
+   *
+   * @param e The event object.
+   */
+  updateText(e: CustomEvent) {
+    this.getInputElement().value = e.detail;
+    this.updateSearchTermValue(e.detail)
+  }
+
+  /**
+   * Returns the searchbox input element
+   */
+  getInputElement(): HTMLInputElement {
+    return this.querySelector('[data-sfx-ref=searchInput]') as HTMLInputElement;
+  }
+
+  /**
+   * Dispatches an `sfx::on_search_box_change` event, and updates the value
+   * property with the payload of the given event.
+   *
+   * Invoked in response to a change to the contents of the searchbox.
+   *
+   * @param e The KeyboardEvent object.
+   */
+  handleChange(e: KeyboardEvent) {
```

Comment:
Now that this is being bound to `oninput`, should we change this to `handleInput` instead?